FlowCrypt/Functionality/Services/GlobalRouter.swift

```diff
@@ -132,6 +136,27 @@ extension GlobalRouter {
         }
     }
 
+    @MainActor func askForContactsPermission(for route: GlobalRoutingType) {
+        logger.logInfo("Ask for contacts permission with \(route)")
+
+        switch route {
+        case .gmailLogin(let viewController):
+            Task {
+                do {
+                    let session = try await googleService.signIn(
+                        in: viewController,
+                        scopes: GeneralConstants.Gmail.contactsScope
+                    )
+                    self.userAccountService.startSessionFor(user: session)
+                } catch {
+                    logger.logInfo("Contacts scope failed with error \(error.errorMessage)")
```

Comment:
If we can, let's use the standard pattern to handle errors in the app which is `Task / do / try / catch / render modal`. Would it be appropriate to render modal here instead of just logging?

Reply:
Yes, you're right, I'll add error modal here